packages/core/src/env-set/index.ts

```diff
@@ -57,18 +60,44 @@ const loadEnvValues = () => {
   });
 };
 
+const throwNotLoadedError = () => {
+  throw new Error(
+    'Env set is not loaded. Make sure to call `await envSet.load()` before using it.'
+  );
+};
+
+/* eslint-disable @silverhand/fp/no-let, @silverhand/fp/no-mutation */
 function createEnvSet() {
-  // eslint-disable-next-line @silverhand/fp/no-let
-  let values = loadEnvValues();
+  let values: Optional<Awaited<ReturnType<typeof loadEnvValues>>>;
+  let pool: Optional<DatabasePoolType>;
 
   return {
-    values,
-    reload: () => {
-      // eslint-disable-next-line @silverhand/fp/no-mutation
-      values = loadEnvValues();
+    get values() {
+      if (!values) {
+        return throwNotLoadedError();
+      }
+
+      return values;
+    },
+    get pool() {
+      if (!pool) {
+        return throwNotLoadedError();
+      }
+
+      return pool;
+    },
+
+    load: async () => {
+      values = await loadEnvValues();
+
+      if (!values.isTest) {
+        const interceptors = [...createInterceptors()];
+        pool = createPool(values.databaseUrl, { interceptors });
```

Comment:
nit: seems that we should not use DB in test env?

maybe?

```ts
      if (!values.isTest || (values.isTest && values.databaseUrl)) {
        const interceptors = [...createInterceptors()];
        pool = createPool(values.databaseUrl, { interceptors });
```

Reply:
this will be replaced in the next PR

Reply:
please check #586